### PR TITLE
admin endpoints

### PR DESCRIPTION
**Pull request by Subham142**
Closes: #313 

## Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.(Specially for Backend)
- [x] My changes generate no new warnings.

## Screenshots

|        Original         |          Updated           |
| :---------------------: | :------------------------: |
| **original screenshot** | <b>updated screenshot </b> |

### CODE REVIEW

File: server/src/controllers/auth.controller.js [hunk 1]
```diff
@@ -29,7 +29,8 @@ const signup = (req, res) => {
             lastName, 
             email, 
             password,
-            confirmPassword
+            confirmPassword,
+            isAmin
```
Comment (himanshujaidka) [inline server/src/controllers/auth.controller.js:33]: causes error

**Review by himanshujaidka — COMMENTED** [review]
pls make another route for the admin `/admin/signup` and similarly for login and make another controller function that handles the functionality for the admin and also the different middleware for the admin route

**Comment by Subham142** [comment]
> other controller function that handles the functionality for the admin and also the different middleware for the admin route

I think there's no need of making other routes and controllers, that's the reason I made the admin middleware, if a user is logged in and is an admin then he/she could visit the routes
If it's not clear, I'll make a controller that returns  all the users for the admin  that might solve your concerns

**Comment by himanshujaidka** [comment]
@codewithvk what do u think?

**Comment by codewithvk** [comment]
@himanshujaidka  Both approaches are right but generally, people make by middleware.

**Pull request merged**